Sources/ZcashLightClientKit/Service/LightWalletGRPCService.swift

```
@@ -454,30 +455,38 @@ extension LightWalletGRPCService: LightWalletServiceNonBlockingAPI {
         }
     }
     
-    public func fetchUTXOsAsync(for tAddress: String, height: BlockHeight) async throws -> [UnspentTransactionOutputEntity] {
+    public func fetchUTXOsAsync(for tAddress: String, height: BlockHeight) async throws -> AsyncThrowingStream<UnspentTransactionOutputEntity, Error> {
         let arg = GetAddressUtxosArg.with { utxoArgs in
             utxoArgs.addresses = [tAddress]
             utxoArgs.startHeight = UInt64(height)
         }
-        var utxos: [UnspentTransactionOutputEntity] = []
-        let stream = compactTxStreamerAsync.getAddressUtxosStream(arg)
-        
-        for try await reply in stream {
-            utxos.append(
-                UTXO(
-                    id: nil,
-                    address: tAddress,
-                    prevoutTxId: reply.txid,
-                    prevoutIndex: Int(reply.index),
-                    script: reply.script,
-                    valueZat: Int(reply.valueZat),
-                    height: Int(reply.height),
-                    spentInTx: nil
-                )
-            )
+
+        let utxoStreamCall = compactTxStreamerAsync.makeGetAddressUtxosStreamCall(arg)
+
+        return AsyncThrowingStream { continuation in
+            Task {
+                continuation.onTermination = { @Sendable _ in
+                    utxoStreamCall.cancel()
+                }
+
+                for try await reply in utxoStreamCall.responseStream {
```

Are we sure the the error in case on `try` will be propagated?

---

UPD: I tried it and without do-catch the error is not propagated